tools/traffic/workers/blob_status_tracker.go

```diff
@@ -0,0 +1,253 @@
+		return true
+	case disperser.BlobStatus_INSUFFICIENT_SIGNATURES:
+		return true
+	case disperser.BlobStatus_CONFIRMED:
+		return true
+	case disperser.BlobStatus_FINALIZED:
+		return true
+	default:
+		return false
+	}
+}
+
+// isBlobStatusConfirmed returns true if the status is a confirmed status.
+func isBlobStatusConfirmed(status disperser.BlobStatus) bool {
+	switch status {
+	case disperser.BlobStatus_CONFIRMED:
+		return true
+	case disperser.BlobStatus_FINALIZED:
+		return true
+	default:
+		return false
+	}
+}
+
+// updateStatusMetrics updates the metrics for the reported status of a blob.
+func (verifier *BlobStatusTracker) updateStatusMetrics(status disperser.BlobStatus) {
+	switch status {
+	case disperser.BlobStatus_UNKNOWN:
+		verifier.unknownCountMetric.Increment()
+	case disperser.BlobStatus_PROCESSING:
+		verifier.processingCountMetric.Increment()
+	case disperser.BlobStatus_DISPERSING:
+		verifier.dispersingCountMetric.Increment()
+	case disperser.BlobStatus_FAILED:
+		verifier.failedCountMetric.Increment()
+	case disperser.BlobStatus_INSUFFICIENT_SIGNATURES:
+		verifier.insufficientSignaturesCountMetric.Increment()
+	case disperser.BlobStatus_CONFIRMED:
+		verifier.confirmedCountMetric.Increment()
+	case disperser.BlobStatus_FINALIZED:
+		verifier.finalizedCountMetric.Increment()
+	default:
+		verifier.logger.Error("unknown blob status", "status:", status)
+	}
+}
+
+// getBlobStatus gets the status of a blob from the disperser service. Returns nil if there was an error
+// getting the status.
+func (verifier *BlobStatusTracker) getBlobStatus(key *UnconfirmedKey) *disperser.BlobStatusReply {
+	ctxTimeout, cancel := context.WithTimeout(*verifier.ctx, verifier.config.GetBlobStatusTimeout)
+	defer cancel()
+
+	status, err := metrics.InvokeAndReportLatency[*disperser.BlobStatusReply](verifier.getStatusLatencyMetric,
+		func() (*disperser.BlobStatusReply, error) {
+			return verifier.disperser.GetBlobStatus(ctxTimeout, key.Key)
+		})
+
+	if err != nil {
+		verifier.logger.Error("failed check blob status", "err:", err)
+		verifier.getStatusErrorCountMetric.Increment()
+		return nil
+	}
+
+	return status
+}
+
+// forwardToReader forwards a blob to the reader. Only called once the blob is ready to be read.
+func (verifier *BlobStatusTracker) forwardToReader(key *UnconfirmedKey, status *disperser.BlobStatusReply) {
+	batchHeaderHash := status.GetInfo().BlobVerificationProof.BatchMetadata.BatchHeaderHash
+	blobIndex := status.GetInfo().BlobVerificationProof.GetBlobIndex()
+
+	confirmationTime := time.Now()
+	confirmationLatency := confirmationTime.Sub(key.SubmissionTime)
+	verifier.confirmationLatencyMetric.ReportLatency(confirmationLatency)
+
+	requiredDownloads := verifier.config.RequiredDownloads
+	var downloadCount int32
+	if requiredDownloads <= 0 {
+		// Allow unlimited downloads.
+		downloadCount = -1
+	} else if requiredDownloads == 0 {
+		// Do not download blob.
+		return
+	} else if requiredDownloads < 1 {
```

Comment:
same with this condition. `requiredDownloads <= 0` includes this and the above conditions?

Reply:
This can be triggered if the configuration is set up with a fractional number for required downloads. Per the original description in the ticket, this allows us to set up a probabilistic download rate, e.g. if we want to download 50% of blobs.

Reply:
> Not sure if it's fixed. If requiredDownloads < 1, it would have been handled on if requiredDownloads < 0 block above

The `requiredDownloads < 1` will trigger if the number of required downloads is between 0 and 1. It's a `float64` value. Any number between 0 and 1 is interpreted as a probabilistic number of downloads.